pkg/service/model_build_load_balancer.go

```diff
@@ -183,28 +183,61 @@ func (t *defaultModelBuildTask) buildLoadBalancerTags(ctx context.Context) (map[
 	return t.buildAdditionalResourceTags(ctx)
 }
 
-func (t *defaultModelBuildTask) buildLoadBalancerSubnetMappings(ctx context.Context, scheme elbv2model.LoadBalancerScheme, ec2Subnets []*ec2.Subnet) ([]elbv2model.SubnetMapping, error) {
+func (t *defaultModelBuildTask) buildLoadBalancerSubnetMappings(_ context.Context, ipAddressType elbv2model.IPAddressType, scheme elbv2model.LoadBalancerScheme, ec2Subnets []*ec2.Subnet) ([]elbv2model.SubnetMapping, error) {
 	var eipAllocation []string
 	eipConfigured := t.annotationParser.ParseStringSliceAnnotation(annotations.SvcLBSuffixEIPAllocations, &eipAllocation, t.service.Annotations)
-	var privateIpv4Addresses []string
-	ipv4Configured := t.annotationParser.ParseStringSliceAnnotation(annotations.SvcLBSuffixPrivateIpv4Addresses, &privateIpv4Addresses, t.service.Annotations)
-
-	// Validation
-	if eipConfigured && ipv4Configured {
-		return []elbv2model.SubnetMapping{}, errors.Errorf("only one of EIP allocations or PrivateIpv4Addresses can be set")
-	}
 	if eipConfigured {
-		if scheme == elbv2model.LoadBalancerSchemeInternal {
-			return []elbv2model.SubnetMapping{}, errors.Errorf("EIP allocations can only be set for internet facing load balancers")
-		} else if len(eipAllocation) != len(ec2Subnets) {
-			return []elbv2model.SubnetMapping{}, errors.Errorf("number of EIP allocations (%d) and subnets (%d) must match", len(eipAllocation), len(ec2Subnets))
+		if scheme != elbv2model.LoadBalancerSchemeInternetFacing {
+			return nil, errors.Errorf("EIP allocations can only be set for internet facing load balancers")
+		}
+		if len(eipAllocation) != len(ec2Subnets) {
+			return nil, errors.Errorf("count of EIP allocations (%d) and subnets (%d) must match", len(eipAllocation), len(ec2Subnets))
 		}
 	}
-	if ipv4Configured {
-		if scheme == elbv2model.LoadBalancerSchemeInternetFacing {
-			return []elbv2model.SubnetMapping{}, errors.Errorf("PrivateIpv4Addresses can only be set for internal balancers")
-		} else if len(privateIpv4Addresses) != len(ec2Subnets) {
-			return []elbv2model.SubnetMapping{}, errors.Errorf("number of PrivateIpv4Addresses (%d) and subnets (%d) must match", len(privateIpv4Addresses), len(ec2Subnets))
+
+	var ipv4Addresses []netip.Addr
+	var rawIPv4Addresses []string
+	ipv4AddrConfigured := t.annotationParser.ParseStringSliceAnnotation(annotations.SvcLBSuffixPrivateIpv4Addresses, &rawIPv4Addresses, t.service.Annotations)
+	if ipv4AddrConfigured {
+		if scheme != elbv2model.LoadBalancerSchemeInternal {
+			return nil, errors.Errorf("PrivateIpv4Addresses can only be set for internal load balancers")
```

Comment:
since this error is visible to the end user, lets mention `private IPv4 addresses` instead

Reply:
addressed :D